kuma/static/styles/components/wiki/_toc.scss

```diff
@@ -126,36 +126,109 @@ On This Page component
     background-color: unset;
     margin: 0;
     padding-top: 0;
+    padding-bottom: 0;
     width: unset;
+
+    @media #{$mq-small-desktop-and-up} {
+        padding-bottom: 20px;
+    }
 }
 
 .document-toc {
-    background-color: $toc-background-color;
-    margin-top: 30px;
-    border: solid 1px $toc-border-color;
+    position: relative;
 
-    &:before {
-        display: block;
-        background-image: linear-gradient(-272deg, $toc-gradient-start-color, $toc-gradient-end-color);
-        content: '';
-        height: 4px;
+    h2 {
+        display: none;
+
+        @media #{$mq-small-desktop-and-up} {
+            display: block;
+        }
     }
 
-    h2 {
-        margin: 15px 0;
-        @include bidi(((padding-left, 10px, padding-right, 0),));
-        font-weight: 400;
-        @include set-font-size(24px);
+    .toc-trigger-mobile {
+        margin-top: $gutter-width / 2;
+        padding: 10px 0;
+        border: 0;
+        border-bottom: 2px solid $blue;
+        color: $link-color;
+        width: 100%;
+        font-weight: normal;
+        text-align: left;
+
+        svg {
+            margin-left: 5px;
+            fill: currentColor;
+            width: 18px;
+            height: 18px;
+            vertical-align: bottom;
+
+            @media #{$mq-small-desktop-and-up} {
+                display: none;
+            }
+        }
+
+        @media #{$mq-small-desktop-and-up} {
+            display: none;
+        }
     }
 
     ul {
-        @include bidi(((padding-left, 20px, padding-right, 0),));
+        display: none;
+        position: absolute;
+        top: 50px;
+        background-color: #fff;
+        padding: 0 20px;
+        border: 2px solid $blue;
+        width: 90%;
```

Comment:
I think something like `width: calc(100% - 40px)` would match the design a little more closely. And then that would  eliminate the need for the `width: 94%;` part below.

Reply:
Good idea, let me give this a whirl.